docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,19 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be used to supply extra context to the ``as_sql``-method
+    for that particular call. This can be useful for Expressions in
+    particular, where ``template`` can be overridden and variables intended
+    for interpolating the ``template``-string can be passed. 3rd-party
+    backends may have different SQL-syntax so they can pass these differences
+    through via ``kwargs``.
+
+    Example::
+
+        self.as_sql(compiler, connection,
+            template="%(function)s(%(arg)s)", function='Print', arg='Hello World'
+        )
```

Comment:
``arg`` in this example is supported because it exists as a placeholder within the ``template`` that we're also passing through. This example was designed to highlight that ``kwargs`` can be interpolated into the ``template`` string.

@timgraham what do you think about the wording and example here? Would a more realistic example be better?